llvm/lib/Target/RISCV/RISCVRVVInitUndef.cpp

```diff
@@ -275,7 +275,7 @@ bool RISCVInitUndef::processBasicBlock(MachineFunction &MF,
       Changed |= handleSubReg(MF, MI, DLD);
     if (MI.isImplicitDef()) {
       auto DstReg = MI.getOperand(0).getReg();
-      if (isVectorRegClass(DstReg))
+      if (DstReg.isVirtual() && isVectorRegClass(DstReg))
```

Comment:
The MachineSink change causes COPY and IMPLICIT_DEF to get merged, creating IMPLICIT_DEF with physical register Def. I'm assuming we don't need to worry about those since they shouldn't be used by a vector instruction.